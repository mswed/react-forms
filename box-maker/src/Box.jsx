import React from 'react';

const Box = ({ width = 100, height = 100, color = 'red', id, remove }) => {
  const handleClick = (evt) => {
    remove(id);
  };
  return (
    <>
      <div
        style={{
          width: `${width}px`,
          height: `${height}px`,
          backgroundColor: color,
          display: 'inline-block',
          margin: '5px',
        }}
      ></div>
      <button onClick={handleClick}>X</button>
    </>
  );
};

export default Box;
