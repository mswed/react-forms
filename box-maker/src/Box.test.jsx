import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import Box from './Box';

it('Renders without crashing', async () => {
  render(<Box />);
});

it('Matches snapshot', async () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});
