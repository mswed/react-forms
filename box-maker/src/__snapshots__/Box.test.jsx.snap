// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 100px; height: 100px; background-color: red; display: inline-block; margin: 5px;"
  />
  <button>
    X
  </button>
</DocumentFragment>
`;
